buildpack/build_test.go

```diff
@@ -27,1335 +27,1325 @@ import (
-									Metadata: map[string]interface{}{"version": "some-version-new"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-							{
-								Require: buildpack.Require{
-									Name:     "some-dep",
-									Metadata: map[string]interface{}{"version": "v1"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-							{
-								Require: buildpack.Require{
-									Name:     "some-replace-version-dep",
-									Metadata: map[string]interface{}{"version": "some-version-new"},
-								},
-								Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-							},
-						},
-						Labels: nil,
-						MetRequires: []string{
-							"some-deprecated-bp-dep",
-							"some-deprecated-bp-replace-version-dep",
-							"some-dep",
-							"some-replace-version-dep",
-						},
-						Processes: nil,
-						Slices:    nil,
-					}); s != "" {
-						t.Fatalf("Unexpected:\n%s\n", s)
-					}
-				})
-
-				it("should convert top level version to metadata.version in the bom", func() {
-					h.Mkfile(t,
-						"[[entries]]\n"+
-							`name = "dep-1"`+"\n"+
-							`version = "v1"`+"\n"+
-							"[[entries]]\n"+
-							`name = "dep-2"`+"\n"+
-							`version = "v2"`+"\n"+
-							"[entries.metadata]\n"+
-							`version = "v2"`+"\n"+
-							"[[entries]]\n"+
-							`name = "dep-3"`+"\n"+
-							"[entries.metadata]\n"+
-							`version = "v3"`+"\n",
-						filepath.Join(appDir, "build-plan-out-A-v1.toml"),
-					)
-
-					br, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv)
-					if err != nil {
-						t.Fatalf("Unexpected error:\n%s\n", err)
-					}
-
-					if s := cmp.Diff(br.LaunchBOM, []buildpack.BOMEntry{
-						{
-							Require: buildpack.Require{
-								Name:     "dep-1",
-								Metadata: map[string]interface{}{"version": "v1"},
-							},
-							Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-						},
-						{
-							Require: buildpack.Require{
-								Name:     "dep-2",
-								Metadata: map[string]interface{}{"version": "v2"},
-							},
-							Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-						},
-						{
-							Require: buildpack.Require{
-								Name:     "dep-3",
-								Metadata: map[string]interface{}{"version": "v3"},
-							},
-							Buildpack: buildpack.GroupElement{ID: "A", Version: "v1"},
-						},
-					}); s != "" {
-						t.Fatalf("Unexpected:\n%s\n", s)
-					}
-				})
-			})
-
-			when("building fails", func() {
-				it("should error when the output buildpack plan is invalid", func() {
-					mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
-					h.Mkfile(t, "bad-key", filepath.Join(appDir, "build-plan-out-A-v1.toml"))
-					if _, err := bpTOML.Build(buildpack.Plan{}, config, mockEnv); err == nil {
-						t.Fatal("Expected error.\n")
-					} else if !strings.Contains(err.Error(), "key") {
-						t.Fatalf("Incorrect error: %s\n", err)
-					}
```

Comment:
Moved to https://github.com/buildpacks/lifecycle/blob/b71e534defe5a2bca5eed1a738f7b8bc52522153/buildpack/build_test.go#L1048-L1055